ecl/hql/hqlgram.y

```diff
@@ -8812,6 +8812,10 @@ simpleDataSet
 
                             bool hasFileposition = getBoolAttributeInList(extra, filepositionAtom, true);
                             record.setown(parser->checkIndexRecord(record, $5, extra));
+
+                            if (!parser->checkAllowed($1, "datafile", "Datafile access"))
+                                extra.setown(createComma(extra.getClear(), createAttribute(_disallowed_Atom)));
```

Comment:
Actually I agree with Jake here.  The original reason why it was done here, rather than at generate time was that it is only in the parser if you know whether something has been signed or not.  However, in a previous pull request for this same change Shamser added a _signed_ attribute to all indexes/dataset to indicate whether it has been signed.  The _disallowed_Atom is repeating that same information.

It should now be easier to check purely at generate time - there would be no need for the parser changes since the information is already available.